lib/rmt/mirror.rb

```diff
@@ -143,32 +134,35 @@ def mirror_data(primary_files, deltainfo_files)
     @downloader.destination_dir = @repository_dir
     @downloader.cache_dir = nil
 
-    failed_downloads = []
-    deltainfo_files.each do |filename|
-      parser = RMT::Rpm::DeltainfoXmlParser.new(
-        File.join(@temp_metadata_dir, filename),
-        @mirror_src
-      )
-      parser.parse
-      to_download = parsed_files_after_dedup(@repository_dir, parser.referenced_files)
-      failed_downloads.concat(@downloader.download_multi(to_download, ignore_errors: true)) unless to_download.empty?
-    end
-
-    primary_files.each do |filename|
-      parser = RMT::Rpm::PrimaryXmlParser.new(
-        File.join(@temp_metadata_dir, filename),
-        @mirror_src
-      )
-      parser.parse
-      to_download = parsed_files_after_dedup(@repository_dir, parser.referenced_files)
-      failed_downloads.concat(@downloader.download_multi(to_download, ignore_errors: true)) unless to_download.empty?
-    end
+    package_files =
+      parse_mirror_data_files(deltainfo_files, RepomdParser::DeltainfoXmlParser) +
+      parse_mirror_data_files(primary_files, RepomdParser::PrimaryXmlParser)
```

Comment:
Would it also make sense to select the needed `package_files` within in method?

Reply:
I followed your suggestions on 687711c.